src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -763,178 +766,166 @@ private static boolean needsConsolidation(final Cigar c) {
-
-        final int numIndels = countIndelElements(cigar);
-        if ( numIndels == 0 )
+        if (cigar.getCigarElements().stream().noneMatch(elem -> elem.getOperator().isIndel())) {
             return cigar;
-        if ( numIndels == 1 )
-            return leftAlignSingleIndel(cigar, refSeq, readSeq, refIndex, readIndex, leftmostAllowedAlignment,true);
+        }
 
-        // if we got here then there is more than 1 indel in the alignment
-        if ( doNotThrowExceptionForMultipleIndels )
-            return cigar;
+        // we need reference bases from the start of the read to the rightmost indel
+        final int lastIndel = IntStream.range(0, cigar.numCigarElements()).filter(n -> cigar.getCigarElement(n).getOperator().isIndel()).max().getAsInt();
+        final int necessaryRefLength = readStart + cigar.getCigarElements().stream().limit(lastIndel + 1).mapToInt(e -> lengthOnReference(e)).sum();
+        Utils.validateArg(necessaryRefLength <= ref.length, "read goes past end of reference");
+
+        // at this point, we are one base past the end of the read.  Now we traverse the cigar from right to left
+        final List<CigarElement> resultRightToLeft = new ArrayList<>();
+        final int refLength = cigar.getCigarElements().stream().mapToInt(e -> lengthOnReference(e)).sum();
+        final IndexRange refIndelRange = new IndexRange(readStart + refLength, readStart + refLength);
+        final IndexRange readIndelRange = new IndexRange(read.length,read.length);
+        for (int n = cigar.numCigarElements() - 1; n >= 0; n--) {
+            final CigarElement element = cigar.getCigarElement(n);
+            // if it's an indel, just accumulate the read and ref bases consumed.  We won't shift the indel until we hit an alignment
+            // block or the read start.
+            if (element.getOperator().isIndel()) {
+                refIndelRange.shiftStartLeft(lengthOnReference(element));
+                readIndelRange.shiftStartLeft(lengthOnRead(element));
+            } else if (refIndelRange.size() == 0 && readIndelRange.size() == 0) {   // no indel, just add the cigar element to the result
+                resultRightToLeft.add(element);
+                refIndelRange.shiftLeft(lengthOnReference(element));
+                readIndelRange.shiftLeft(lengthOnRead(element));
+            } else {    // there's an indel that we need to left-align
+                // we can left-align into match cigar elements but not into clips
+                final int maxShift = element.getOperator().isAlignment() ? element.getLength() : 0;
+                final Pair<Integer, Integer> shifts = leftAlignAlleles(Arrays.asList(ref, read), Arrays.asList(refIndelRange, readIndelRange), maxShift);
+
+                // account for new match alignments on the right due to left-alignment
+                resultRightToLeft.add(new CigarElement(shifts.getRight(), CigarOperator.MATCH_OR_MISMATCH));
```

Comment:
It's different here because below we're re-emitting a shorter version of a cigar element that already existed.  Here we're creating something new resulting from the left-alignment.